plan/expression_rewriter.go

```diff
@@ -20,6 +20,24 @@ import (
 // EvalSubquery evaluates incorrelated subqueries once.
 var EvalSubquery func(p PhysicalPlan, is infoschema.InfoSchema, ctx context.Context) ([]types.Datum, error)
 
+// GetInfoSchema gets infoschema from ctx.
+var GetInfoSchema func(ctx context.Context) infoschema.InfoSchema
+
+// EvalAstExpr evaluates ast expression directly.
+func EvalAstExpr(expr ast.ExprNode, ctx context.Context) (types.Datum, error) {
+	b := &planBuilder{
+		ctx:       ctx,
+		allocator: new(idAllocator),
+		colMapper: make(map[*ast.ColumnNameExpr]int),
+		is:        GetInfoSchema(ctx),
```

Comment:
You can use `is: ctx.GetSessionVars().TxnCtx.InfoSchema.(infoschema.InfoSchema)` directly.